app/data_models/metadata_proxy.py

```diff
@@ -0,0 +1,43 @@
+from typing import Any, Mapping, Optional
+
+
+class MetadataProxy:
+    def __init__(self, metadata: Optional[Mapping[str, Any]]):
+        self.metadata: Optional[Mapping[str, Any]] = metadata
```

Comment:
Does this need type hinting again since done on line 5?

Reply:
Removed this